test/e2e/framework/.import-restrictions

```diff
@@ -1,269 +1,34 @@
 rules:
-  - selectorRegexp: k8s[.]io/kubernetes/pkg/
+  # The core E2E framework is meant to be a normal Kubernetes client,
+  # which means that it shouldn't depend on internal
+  # code. But we are not there yet, so some exceptions
+  # have to be allowed. Over time the list of allowed
+  # packages should get shorter, not longer.
```

Comment:
I agree with the above comment.
We tried to solve the dependency issue so hardly, but we saw that is hard to do that actually.

Reply:
Almost done now :grin: 

The last remaining issue is embedding code that is only used by test/e2e_node inside test/e2e/framework (mostly for the TestContext). That pulls in the k/k/pkg dependency.